lib/auth/auth_with_roles.go

```diff
@@ -228,6 +228,18 @@ func HasBuiltinRole(checker services.AccessChecker, name string) bool {
 	return true
 }
 
+// HasRemoteBuiltinRole checks the type of the role set returned and the name.
+// Returns true if role set is remote builtin and the name matches.
+func HasRemoteBuiltinRole(checker services.AccessChecker, name string) bool {
+	if _, ok := checker.(RemoteBuiltinRoleSet); !ok {
+		return false
+	}
+	if !checker.HasRole(name) {
+		return false
+	}
+	return true
```

Comment:
```suggestion
	_, isRemote := checker.(RemoteBuiltinRoleSet)
	return isRemote && checker.HasRole(name)
```

I don't really care, but what do you think about expressing it this way instead?